chainer/variable.py

```diff
@@ -1143,21 +1160,22 @@ def _to_device(self, device, allow_unchaining):
         elif not was_chainerx and is_chainerx:
             self._chainerx_name = self._node.name
 
+        # Note: It affects the whole behavior of Variable's methods.
+        #       We must place them in this place.
         self._device = device
         self._has_chainerx_array = is_chainerx
 
-        if arr is None:
-            return
-
-        if backend.get_device_from_array(arr) == device:
+        if arr is None or old_device == device:
```

Comment:
How about checking `old_device == device` earlier?

Reply:
It depends on the conclusion of #6116 because `Parameter` expects `self._device` has a valid configuration even if `arr is None`. We can move `old_device == device` independently but I’d like to wait until the discussion is finished.

Reply:
I'd like to leave it because #6116 has not yet been concluded.

Reply:
OK.